tmt/package_managers/dnf.py

```diff
@@ -165,6 +176,24 @@ def install_debuginfo(
 
         return script
 
+    def install_repository(self, repository: Repository) -> ShellScript:
+        repo_path = f"/etc/yum.repos.d/{repository.filename}"
+        return ShellScript(
+            rf"""
+            {self.guest.facts.sudo_prefix} tee {repo_path} <<'EOF'
```

Comment:
I do not think we need to quote `EOF` here

```suggestion
            {self.guest.facts.sudo_prefix} tee {repo_path} <<EOF
```

Reply:
yum repository configurations that may include patterns like $baseurl or other placeholders that should not be interpreted as shell variables. Using <<'EOF' ensures no unintended shell variable substitution or command expansion occurs, preserving the exact content of `repository.content`.

Reply:
Discussed in review meeting.